Fushigi/ui/widgets/CourseScene.cs

```diff
@@ -2335,6 +2405,42 @@ private async Task<(CourseActor?, KeyboardModifier modifiers)> PickLinkDestInVie
             batchAction.Commit($"{IconUtil.ICON_TRASH} {actionName}");
         }
 
+        //TODO making this undoable
+        private async Task DeleteLayerWithWarningPrompt(string layer,
+            CourseActorHolder actorArray, CourseAreaEditContext ctx)
+        {
+            var actors = actorArray.mActors.FindAll(x => x.mLayer == layer);
+            var confirm = await OperationWarningDialog.ShowDialog(mPopupModalHost,
+                "Deletion warning",
+                "Are you sure you want to delete " +
+                $"{layer}");
+
+            if (confirm == OperationWarningDialog.DialogResult.Cancel)
+                return;
```

Comment:
There's no need to display 2 OperationWarningDialogs, please **remove** this.
If there is no warnings simply display a Dialog without them (you could probably just move this code into the `else` block of `if (!noWarnings)` and ideally revert the if for easier readablity)